### PR TITLE
fix: retries for hub client

### PR DESCRIPTION
**Pull request by mtsitrin**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Close https://github.com/dymensionxyz/dymint/issues/551

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: settlement/dymension/dymension.go [hunk 1]
```diff
@@ -508,11 +551,11 @@ func (d *HubClient) waitForBatchInclusion(batchStartHeight uint64) (*settlement.
...
-			resultRetriveBatch = latestBatch
-			return nil
+		if latestBatch.Batch.StartHeight != batchStartHeight {
+			return settlement.ErrBatchNotFound
 		}
-		return settlement.ErrBatchNotFound
+		resultRetriveBatch = latestBatch
+		return nil
 	}, retry.Context(d.ctx), retry.LastErrorOnly(true),
 		retry.Delay(d.batchRetryDelay), retry.Attempts(d.batchRetryAttempts), retry.MaxDelay(batchRetryMaxDelay))
```
Comment (danwt) [inline settlement/dymension/dymension.go:560]: this code is duplicated a lot
can we dry out by extracting d.retry(fn) which uses these default args?

**Review by danwt — CHANGES_REQUESTED** [review]

**Comment by danwt** [comment]
It seems GetSequencers and GetBatchAtIndex dont require the rollapp id as args

**Review by omritoptix — COMMENTED** [review]
LGTM. we can dry out as mentioned by @danwt.

**Comment by mtsitrin** [comment]
> It seems GetSequencers and GetBatchAtIndex dont require the rollapp id as args

yes it requires refactoring (https://github.com/dymensionxyz/dymint/issues/629)

File: block/manager.go [hunk 2]
```diff
@@ -279,23 +278,5 @@ func (m *Manager) applyBlockCallback(event pubsub.Message) {
...
-		retry.LastErrorOnly(true),
-		retry.Context(ctx),
-		retry.Attempts(1),
-	)
-	if err != nil {
-		return resultRetrieveBatch, err
-	}
-	return resultRetrieveBatch, nil
+	return m.settlementClient.RetrieveBatch()
 }
```
Comment (danwt) [inline block/manager.go:282]: this function could be inlined, it seems

**Review by danwt — APPROVED** [review]

File: block/manager.go (hunk 2 above)
Comment (danwt) [reply block/manager.go:282]: approved anyway

Comment (mtsitrin) [reply block/manager.go:282]: right. done in other PR

**Pull request merged**